pkg/storegateway/chunkscache/memcached.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/thanos-io/thanos/blob/main/pkg/store/cache/memcached.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Thanos Authors.
+
+package chunkscache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/cache"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/grafana/mimir/pkg/util/pool"
+)
+
+const (
+	memcachedDefaultTTL = 7 * 24 * time.Hour
+)
+
+// MemcachedChunksCache is a memcached-based index cache.
+type MemcachedChunksCache struct {
+	logger    log.Logger
+	memcached cache.MemcachedClient
```

Comment:
`cache.MemcachedClient` is an alias for `cache.RemoteCacheClient`.

Reply:
are you saying we should use `cache.RemoteCacheClient` instead? Why do we even have two 😄 ?

Reply:
yeah, we should use `RemoteCacheClient`. We have two because Reasons